From 19d5d0559a812ae321956d6d9329f1032fbc6b66 Mon Sep 17 00:00:00 2001
From: James Haggerty <james.haggerty@morsemicro.com>
Date: Wed, 20 Aug 2025 10:45:04 +1000
Subject: [PATCH] dpp: (APP-5117) remove configuration hack in
 gas_request_handler

When originally implemented, this was put in presumably because
either (a) dppd wasn't provided the correct config or
(b) dppd wasn't providing an updated password.

For OpenWrt, we restart dppd and tell it the password every time our
config changes, so this is unnecessary. More importantly, it also
removes the entire existing conf_sta, which means it removes any
custom configuration (conf_extra_name/value) we send via
dpp_custom_configuration, which is used for HL2.

For details, see SW-16596.
---
 src/ap/dpp_hostapd.c | 55 --------------------------------------------
 1 file changed, 55 deletions(-)

diff --git a/src/ap/dpp_hostapd.c b/src/ap/dpp_hostapd.c
index 4358761d7..e2388cc75 100644
--- a/src/ap/dpp_hostapd.c
+++ b/src/ap/dpp_hostapd.c
@@ -3152,12 +3152,6 @@ hostapd_dpp_gas_req_handler(struct hostapd_data *hapd, const u8 *sa,
 {
 	struct dpp_authentication *auth = hapd->dpp_auth;
 	struct wpabuf *resp;
-	int akm;
-	const char *password = NULL;
-	bool sae = false;
-#ifdef CONFIG_SAE
-	struct sae_password_entry *e;
-#endif /* CONFIG_SAE */
 
 	wpa_printf(MSG_DEBUG, "DPP: GAS request from " MACSTR, MAC2STR(sa));
 	if (!auth || (!auth->auth_success && !auth->reconfig_success) ||
@@ -3197,55 +3191,6 @@ hostapd_dpp_gas_req_handler(struct hostapd_data *hapd, const u8 *sa,
 	wpa_msg(hapd->msg_ctx, MSG_INFO, DPP_EVENT_CONF_REQ_RX "src=" MACSTR,
 		MAC2STR(sa));
 
-	/* Update station configuration to match the AP requirements */
-	akm = dpp_akm_from_hapd_wpa_key(hapd->conf->wpa_key_mgmt);
-	if (akm == DPP_AKM_UNKNOWN) {
-		wpa_printf(MSG_DEBUG,
-			"DPP: unsupported AKM for DPP flow, using connector as default");
-		akm = DPP_AKM_DPP;
-	}
-
-	/* Free and reallocate existing station configuration */
-	if (auth->conf_sta)
-		dpp_configuration_free(auth->conf_sta);
-
-	auth->conf_sta =
-		dpp_configuration_alloc(dpp_akm_str(akm));
-
-	auth->conf_sta->netrole = DPP_NETROLE_STA;
-
-	os_memcpy(auth->conf_sta->ssid,
-		  hapd->conf->ssid.ssid,
-		  hapd->conf->ssid.ssid_len);
-	auth->conf_sta->ssid_len = hapd->conf->ssid.ssid_len;
-
-#ifdef CONFIG_SAE
-	if (hapd->conf->wpa & WPA_PROTO_RSN)
-		sae = hapd->conf->wpa_key_mgmt & WPA_KEY_MGMT_SAE;
-#endif /* CONFIG_SAE */
-
-	if (dpp_akm_psk(akm) || dpp_akm_sae(akm)) {
-#ifdef CONFIG_SAE
-		for (e = hapd->conf->sae_passwords; sae && e && !password;
-			e = e->next) {
-			if (e->identifier || !is_broadcast_ether_addr(e->peer_addr))
-				continue;
-			password = e->password;
-		}
-#endif /* CONFIG_SAE */
-		if (!password && hapd->conf->ssid.wpa_passphrase_set &&
-		    hapd->conf->ssid.wpa_passphrase) {
-			password = hapd->conf->ssid.wpa_passphrase;
-		}
-
-		if (!password) {
-			wpa_printf(MSG_INFO, "DPP: SAE passphrase is missing");
-			return NULL;
-		}
-
-		auth->conf_sta->passphrase = os_strdup(password);
-	}
-
 	resp = dpp_conf_req_rx(auth, query, query_len);
 	if (!resp)
 		wpa_msg(hapd->msg_ctx, MSG_INFO, DPP_EVENT_CONF_FAILED);
-- 
2.34.1

